app/models/statistics_report/domains_data.rb

```diff
@@ -0,0 +1,37 @@
+class StatisticsReport
+  class DomainsData
+    attr_reader :start_date
+    attr_reader :end_date
+    ATTRS = %i[unregistered_domains_count_by_end
+               registered_domains_count_by_end].freeze
+
+    attr_accessor(*ATTRS)
+
+    def initialize(start_date:, end_date:)
+      @start_date = start_date
+      @end_date = end_date
+      ATTRS.each { |attr| send("#{attr}=", {}) }
+    end
+
+    def gather_data
+      (start_date..end_date).each do |date|
+        unregistered_domains_count_by_end[date] = unregistered_domains[date]&.count.to_i
+        registered_domains_count_by_end[date] = registered_domains[date]&.count.to_i
+      end
+    end
+
+    def unregistered_domains
+      Result.unregistered
```

Comment:
Wouldn't it be better to move this query to Result model scope / method?

Reply:
I prefer not to clog model scopes with anything what used just in one place.